src/pools.jl

```diff
@@ -51,6 +51,35 @@ safesizehint!(x, n) = sizehint!(x, min(4096, n))
 # determines whether we'll look up object caches in .keyedvalues or .values
 iskeyed(::Pool{K}) where {K} = K !== Nothing
 
+Base.keytype(::Type{<:Pool{K}}) where {K} = K
+Base.keytype(p::Pool) = keytype(typeof(p))
+
+Base.valtype(::Type{<:Pool{<:Any, T}}) where {T} = T
+Base.valtype(p::Pool) = valtype(typeof(p))
+
+"""
+    Pools.max(pool::Pool) -> Int
+
+Return the maximum number of objects permitted to be in use at the same time.
+See `Pools.permits(pool)` for the number of objects currently in use.
+"""
+max(pool::Pool) = Base.@lock pool.lock pool.max
```

Comment:
i didn't go for "in use" ("usage", "reusable", etc) because although that's a good description for the HTTP use-case, this Pool functionality is pretty generic (and we already have punny names like `drain!` and the docs talk about "permits")

But also i didn't like `capacity` because that suggested to me that there's a limit on number of "in the pool" rather than "in use".

happy to change to "use"-type names if you think that's better